chrome/model.js

```diff
@@ -82,70 +76,72 @@ GmailToTrello.Model.prototype.makeAvatarUrl = function(avatarHash) {
 
+    self.event.fire('onBeforeLoadPopup');
+
+    self.trello.user = null;
+    self.trello.boards = null;
+
     // get user's info
-    gtt_log('loadTrelloData: User info');
+    gtt_log('loadPopupData: User info');
     Trello.get('members/me', {}, function(data) {
         if (!data || !data.hasOwnProperty('id')) {
             return false;
         }
-
         self.trello.user = data;
-
-        gtt_log('loadTrelloData: User boards');
-        self.trello.boards = null;
-        Trello.get('members/me/boards', {
-                'organization': 'true',
-                'organization_fields': 'displayName',
-                'filter': 'open',
-                'fields': 'name' /* "name,closed" */
-            }, function(data) {
-                var validData = Array();
-                for (var i = 0; i < data.length; i++) {
-                    // if (data[i].idOrganization === null)
-                    //   data[i].idOrganization = '-1';
-
-                    // Only accept opening boards
-                    if (i==0) {
-                        // gtt_log(JSON.stringify(data[i]));
-                    }
-                    if (data[i].closed != true) {
-                        validData.push(data[i]);
-                    }
-                }
-                // gtt_log('loadTrelloData: Boards data');
-                // gtt_log(JSON.stringify(data));
-                // gtt_log(JSON.stringify(validData));
-                self.trello.boards = validData;
-                self.checkTrelloDataReady();
-            }, function failure(data) {
-            self.event.fire('onAPIFailure', {data:data});
-            }
-        );
-        self.checkTrelloDataReady();
+        self.checkPopupDataReady();
     }, function failure(data) {
         self.event.fire('onAPIFailure', {data:data});
     });
+
+    // get user's boards
+    gtt_log('loadPopupData: User boards');
+    Trello.get('members/me/boards', {
+            'organization': 'true',
+            'organization_fields': 'displayName',
+            'filter': 'open',
+            'fields': 'name' /* "name,closed" */
+        }, function(data) {
+            var validData = Array();
+            for (var i = 0; i < data.length; i++) {
+                // if (data[i].idOrganization === null)
+                //   data[i].idOrganization = '-1';
+
+                // Only accept opening boards
+                if (i==0) {
+                    // gtt_log(JSON.stringify(data[i]));
+                }
+                if (data[i].closed != true) {
+                    validData.push(data[i]);
+                }
+            }
+            // gtt_log('loadPopupData: Boards data');
+            // gtt_log(JSON.stringify(data));
+            // gtt_log(JSON.stringify(validData));
+            self.trello.boards = validData;
+            self.checkPopupDataReady();
+        }, function failure(data) {
+            self.event.fire('onAPIFailure', {data:data});
+        }
+    );
+
```

Comment:
This section above is actually unnecessary. I changed it as part of an earlier commit. But I actually think it's a bit better this way. It's just loading /members/me and /members/me/boards in parallel rather than sequentially. Since you're using checkPopupDataReady() to take the next step, it's fine to run them in parallel. Should be a bit faster.